src/containers/ProductPage/ProductPage.jsx

```diff
@@ -0,0 +1,154 @@
+    try {
+      const data = await getItemById(id);
+
+      getGallery(data);
+      setProduct(data);
+    } catch (err) {
+      setError(err.message);
+    }
+  };
+
+  const handleChangeSelectedProduct = (idx) => {
+    setSelectedProduct(idx);
+    setGalleryIndex(idx);
+  };
+
+  const handleAddItemToCart = async () => {
+    const { price_per_unit } = product;
+    const data = {
+      price_per_unit,
+      quantityToPurchase: 1,
+      img: product.variants[selectedProduct].img,
+      quantity: product.variants[selectedProduct].quantity,
+      name: product.variants[selectedProduct].variant_name,
+      item_id: product.id,
+    };
+
+    const cartItem = await getCartItemByName(data.name);
+    try {
+      if (data.quantity === 0) {
+        throw new Error(`${cartItem.name} is out of stock`);
+      }
+
+      if (cartItem && cartItem.quantityToPurchase < cartItem.quantity) {
+        await increaseCartItemQuantity(data.name);
+        refreshCart();
+        return;
+      } else if (
+        cartItem?.hasOwnProperty("quantity") &&
+        cartItem?.quantityToPurchase === cartItem?.quantity
+      ) {
+        throw new Error(`Sorry, we have ${cartItem.quantity || 0} remaining`);
+      }
+      await addItemToCart(data, data.name);
+      refreshCart();
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const hatVariation =
+    product?.category === "hat" &&
+    product.variants.map((item, idx) => (
+      <div
+        className={styles.HatVariation}
+        key={idx}
+        onClick={() => handleChangeSelectedProduct(idx)}
+      >
+        <img src={item.img} alt="" />
+        {item.quantity === 0 && <SoldOut />}
+      </div>
+    ));
+  const hoddieVariation =
+    product?.category === "hoodie" &&
+    product.variants.map((item, idx) => (
+      <button key={idx} onClick={() => handleChangeSelectedProduct(idx)}>
+        <span>{item.size}</span>
+      </button>
+    ));
+  const handleFavouriteToggle = async () => {
+    await updateItemFavourite(id, product.favourited);
+    setRefresh(refresh + 1);
+  };
+
+  return (
+    <div>
+      {product ? (
+        <div className={styles.ProductPage}>
+          <div className={styles.ImgContainer}>
+            {productGallery ? (
+              <>
+                <img src={productGallery[galleryIndex]} alt="" />
+              </>
+            ) : (
+              <p>Image unavaliable</p>
+            )}
+            {product.variants[selectedProduct].quantity === 0 && <SoldOut />}
+          </div>
+          <div className={styles.DetailsContainer}>
+            <h2>{product.name}</h2>
+            <span>${product.price_per_unit}</span>
+            <div className={styles.ProductPage_Variants}>
+              {hatVariation}
+              {hoddieVariation}
+            </div>
+            <div>
+              <button
+                disabled={product.variants[selectedProduct].quantity === 0}
+                onClick={handleAddItemToCart}
```

Comment:
Give user some feedback when a product gets successfully added to cart, for example using [Toast Notifications](https://www.geeksforgeeks.org/reactjs-toast-notification/)